version_test.go

```diff
@@ -0,0 +1,175 @@
+// Copyright 2023 Mineiros GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package terramate_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/mineiros-io/terramate"
+	"github.com/mineiros-io/terramate/errors"
+	errtest "github.com/mineiros-io/terramate/test/errors"
+)
+
+func TestTerramateVersionConstraints(t *testing.T) {
+	t.Parallel()
+
+	type testcase struct {
+		version    string
+		constraint string
+		want       error
+	}
+
+	for _, tc := range []testcase{
+		{
+			version:    "0.0.0",
+			constraint: "0.0.0",
```

Comment:
@mariux this is currently valid but it's not with `apparentlymart/go-versions` package:
https://github.com/apparentlymart/go-versions#unspecified-versions
So when user provides `terramate.required_version_allow_pre_releases` it will fail.
Is it ok?